### PR TITLE
CSHARP-2418: Add test that reads ignore db/collection readConcern in transaction

### PR DESCRIPTION
**Pull request by vincentkam**
https://jira.mongodb.org/browse/CSHARP-2418
 https://evergreen.mongodb.com/version/5c74532fc9ec441d67551138

### CODE REVIEW

**Comment by vincentkam** [comment]
Ready for review!

File: tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs [hunk 1]
```diff
@@ -65,23 +67,29 @@ protected override void CallMethod(CancellationToken cancellationToken)
         {
             if (_session == null)
             {
-                _result = _database.RunCommand<BsonDocument>(_command, _readPreference, cancellationToken);
+                _result = _database.WithReadConcern(_readConcern).RunCommand<BsonDocument>(_command, _readPreference, cancellationToken);
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs]: Make multiline to be consistent with your other changes?

File: tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs [hunk 2]
```diff
@@ -92,6 +100,12 @@ protected override void SetArgument(string name, BsonValue value)
                 case "command":
                     _command = value.AsBsonDocument;
                     return;
+                
+                case "databaseOptions":
+                    _readConcern = value.AsBsonDocument.Contains("readConcern")
+                        ? ReadConcern.FromBsonDocument(value["readConcern"].AsBsonDocument)
+                        : _readConcern;
+                    return;
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs:111]: I think this use of ternary is crazy.

I'd rather see something like:

    if (value.AsBsonDocument.TryGetValue("readConcern", out BsonValue readConcernValue)
    {
        _readConcern = ReadConcern.FromBsonDocument(readConcernValue.AsBsonDocument);
    }
    return;

**Review by rstam — CHANGES_REQUESTED** [review]

File: tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs (hunk 1 above)
Comment (vincentkam) [reply tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs]: Fixed!

File: tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs (hunk 2 above)
Comment (vincentkam) [reply tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs:111]: Okay, I've taken your changes with a minor tweak!

Comment (rstam) [reply tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs:111]: Oh nice... I didn't know you could use var like that.

**Review by rstam — APPROVED** [review]

File: tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs (hunk 2 above)
Comment (vincentkam) [reply tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenRunCommandTest.cs:111]: 😺

**Comment by vincentkam** [comment]
Merged, squashed and pushed!

**Pull request closed without merging**